### PR TITLE
Mrazadar/PON-124/Subscription-checkout-without-client-secret

### PR DESCRIPTION
**Pull request by mrazadar**
This pull request renders the Stripe payment elements without the need of client-secret. And uses `elements_enable_deferred_intent_beta_1` to render the form for subscriptions. 

Screen Shot 2023-04-07 at 12 55 28 AM

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -32,4 +32,5 @@ APPLE_PAY_MERCHANT_CAPABILITIES='supports3DS,supportsCredit,supportsDebit',
 STRIPE_API_VERSION=2022-08-01;server_side_confirmation_beta=v1
 STRIPE_BETA_FLAG=server_side_confirmation_beta_1
 STRIPE_PUBLISHABLE_KEY=null
+STRIPE_DEFERRED_INTENT_BETA_FLAG=elements_enable_deferred_intent_beta_1
```
Comment (julianajlk) [inline .env.development:35]: Just a reminder to do the `edx-internal` changes for this beta flag as well, for stage and prod yml files

File: src/subscription/checkout/StripeOptions.js [hunk 2]
```diff
@@ -3,9 +3,18 @@
...
-  clientSecret: clientSecretId,
+  mode: 'subscription',
+  /**
+   * TODO: this should be fetched dynamically
+   * NOTE: this is important to set the right subscription price as defined within stripe subscription
+   * otherwise card details will not load properly
+   * https://dashboard.stripe.com/test/subscriptions/sub_1Mtd3sH4caH7G0X1dvzlMT3o
+   * */
+  amount: 55.00,
+  currency: 'usd', // TODO: this should be fetched dynamically
```
Comment (julianajlk) [inline src/subscription/checkout/StripeOptions.js:16]: This is available in `GET /stripe-checkout` that Chris merged recently, but I noticed it's `'USD'` which it gets from Discovery, I wonder if capitalized will be an issue

File: .env.development (hunk 1 above)
Comment (julianajlk) [reply .env.development:35]: Ok just saw your TODO comment in another file, thank you!

File: src/subscription/checkout/SubscriptionCheckout.jsx [hunk 3]
```diff
@@ -32,18 +31,17 @@ export const SubscriptionCheckout = () => {
     isSubscriptionDetailsProcessing,
     paymentMethod,
   } = useSelector(subscriptionSelector);
-  const { clientSecretId, isClientSecretProcessing } = useSelector(getClientSecretSelector);
+  // const { clientSecretId, isClientSecretProcessing } = useSelector(getClientSecretSelector);
```
Comment (julianajlk) [inline src/subscription/checkout/SubscriptionCheckout.jsx:34]: You still want to keep this commented out?

File: src/subscription/checkout/SubscriptionCheckout.jsx [hunk 4]
```diff
@@ -72,28 +70,26 @@ export const SubscriptionCheckout = () => {
...
+      {
+        !loading ? (
+          <>
+            <Elements options={options} stripe={stripePromise}>
+              <StripePaymentForm
+                options={options}
+                onSubmitPayment={handleSubmitStripe}
+                onSubmitButtonClick={handleSubmitStripeButtonClick}
+                isProcessing={stripeIsSubmitting}
+                isQuantityUpdating={isQuantityUpdating}
```
Comment (julianajlk) [inline src/subscription/checkout/SubscriptionCheckout.jsx:84]: We don't need this for subscriptions, there is no quantity form that is updating (this is used in regular flow for bulk purchases)

File: src/subscription/checkout/SubscriptionCheckout.test.jsx [hunk 5]
```diff
@@ -85,7 +83,7 @@ jest.mock('react-tooltip/node_modules/uuid', () => ({
 
 jest.mock('./StripeOptions', () => ({
   getStripeOptions: jest.fn().mockReturnValue({
-    clientSecret: 'secretKey',
+    // clientSecret: 'secretKey',
```
Comment (julianajlk) [inline src/subscription/checkout/SubscriptionCheckout.test.jsx:86]: Need to keep?

**Review by julianajlk — APPROVED** [review]
LGTM!

File: src/subscription/checkout/StripeOptions.js (hunk 2 above)
Comment (mrazadar) [reply src/subscription/checkout/StripeOptions.js:16]: added `toLowerCase` transformation here -> in my next PR.

File: src/subscription/checkout/SubscriptionCheckout.jsx (hunk 3 above)
Comment (mrazadar) [reply src/subscription/checkout/SubscriptionCheckout.jsx:34]: removed

Comment (mrazadar) [reply src/subscription/checkout/SubscriptionCheckout.jsx:34]: in next follow up PR.

File: src/subscription/checkout/SubscriptionCheckout.jsx (hunk 4 above)
Comment (mrazadar) [reply src/subscription/checkout/SubscriptionCheckout.jsx:84]: removed

File: src/subscription/checkout/SubscriptionCheckout.test.jsx (hunk 5 above)
Comment (mrazadar) [reply src/subscription/checkout/SubscriptionCheckout.test.jsx:86]: removed

**Pull request merged**